### PR TITLE
Generic gql hook

### PR DESCRIPTION
**Pull request by yonas-seyoum**

### CODE REVIEW

**Review by leulsege — APPROVED** [review]

**Review by leulsege — COMMENTED** [review]
As a comment, we use custom hook for reusability yeah, if that so how do see your custom hook from that point, is that generic

File: applications/webapp/src/hooks/configs.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import { GET_USER_QUERY } from "./query";
+import { userSchema } from "./schema";
+import { QueryConfigs, UserData } from "./types";
+
+export const queryConfigs: QueryConfigs = {
+  GET_USER: {
+    query: GET_USER_QUERY,
+    schema: userSchema,
+    data: {} as UserData,
```
Comment (peterrogov) [inline applications/webapp/src/hooks/configs.ts:21]: Not sure what this is supposed to do, but looks strange

File: applications/webapp/src/hooks/useGenericQuery.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    throw new Error(`Unsupported query type: ${queryType}`);
+  }
+
+  const { data, loading, error } = useQuery(config.query, { variables });
+
+  if (data) {
+    const validationResult = config.schema.validate(data);
+    if (validationResult.error) {
+      throw new Error(
+        `Invalid data structure returned by the query. ${validationResult.error}`,
```
Comment (peterrogov) [inline applications/webapp/src/hooks/useGenericQuery.ts:37]: How about "Unexpected response"?

**Review by peterrogov — APPROVED** [review]
Overall is ok. A few minor comments I suggest fixing. Also I believe there's no reason to have separate LoadingOverlay and ErrorOverlay. I would have just one overlay and make it really and overlay. If loading, then it shows loading status, if error, then it displays error. If loading complete, shows children

**Pull request merged**